File: 2020/day-3/part1.py
#!/usr/bin/env python3

SHIFT_X = 3
SHIFT_Y = 1

filepath = 'input.txt'
map = open(filepath).read().split('\n')

height = len(map)
width = len(map[0])

x = 0
y = 0

trees_cnt = 0

while y < height:
  if map[y][x] == '#':
    trees_cnt += 1
  x = (x + SHIFT_X) % width
  y += SHIFT_Y

print(trees_cnt)

File: 2021/day-2/part1.rb
#!/usr/bin/env ruby

x, y = 0, 0
data = File.open('input.txt').read.split("\n").each do |line|
  command, change = line.split
  change = change.to_i
  case command
  when "forward" then x += change
  when "down"    then y += change
  when "up"      then y -= change
  end
end

puts x*y